extensions/charts/verify.go

```diff
@@ -37,3 +37,67 @@ func VerifyChartInstall(client *catalog.Client, chartNamespace, chartName string
 	})
 	return err
 }
+
+// VerifyChartUpgrade verifies that the app from a chart was successfully upgraded
+func VerifyChartUpgrade(client *catalog.Client, chartNamespace, chartName, chartVersion string) error {
+	watchAppInterface, err := client.Apps(chartNamespace).Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + chartName,
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	if err != nil {
+		return err
+	}
+
+	err = wait.WatchWait(watchAppInterface, func(event watch.Event) (ready bool, err error) {
+		app := event.Object.(*v1.App)
+
+		state := app.Status.Summary.State
+		if state == string(v1.StatusPendingUpgrade) {
+			return true, nil
+		}
+
+		if state == string(v1.StatusFailed) {
+			return false, errors.New("chart upgrade has failed")
+		}
+		return false, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	watchAppInterface, err = client.Apps(chartNamespace).Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + chartName,
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	if err != nil {
+		return err
+	}
+
+	err = wait.WatchWait(watchAppInterface, func(event watch.Event) (ready bool, err error) {
+		app := event.Object.(*v1.App)
+
+		state := app.Status.Summary.State
+		if state == string(v1.StatusDeployed) {
+			if app.Spec.Chart.Metadata.Version == chartVersion {
+				return true, nil
+			}
+			return false, errors.New("chart not upgraded to expected version: " + chartVersion)
+		}
+
+		if state == string(v1.StatusFailed) {
+			return false, errors.New("chart upgrade has failed")
+		}
+		return false, nil
+	})
+	return err
+}
+
+// GetExistingChartInstall returns the App object of an installed chart.
+func GetExistingChartInstall(client *catalog.Client, clusterName, chartName, chartNamespace string) (*v1.App, error) {
+	appObject, err := client.Apps(chartNamespace).Get(context.TODO(), chartName, metav1.GetOptions{})
```

Comment:
I am unsure why we need this for a single client wrap, what's the pros/cons of adding this function?

Reply:
I definitely agree on this. Is this really needed. The client call is already a single liner, this just seems like more bloat.

Reply:
sure, updated / rm'd from public funciton list